assets/files/etc/kubernetes/manifests/keepalived.yaml

```diff
@@ -94,7 +94,7 @@ spec:
   - name: keepalived
     securityContext:
       privileged: true
-    image: quay.io/celebdor/keepalived:latest
+    image: registry.svc.ci.openshift.org/ocp/4.2@sha256:daa9f390c43563b67546cd5b4cf3d8e351c3530f8091f523a73061fa441e8818
```

Comment:
Obviously I need to figure out how to reference this image properly.

Reply:
You won't be able to do this "properly" in dev-scripts.  I expect these manifests will move to MCO, right?

There's a bit of magic here you need to hook into.  Assuming MCO is what has this manifest, then the image needs to be listed in this file as well: https://github.com/openshift/machine-config-operator/blob/master/install/image-references

By being listed in `image-references` for one of the cluster operators, the image will get included in the release payload.  There's also magic in CVO to convert image references in CVO deployed manifests to use the right image for that release payload.  It looks like MCO has this manifest that declares a config map of image references.  I would expect keepalived to be added here as well, so you can get the right image from your MCO code:  https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_02_images.configmap.yaml

Reply:
Yeah this will need to go into the MCO PR @bcrochet is working on, when that lands all this asset mangling will go away ref https://github.com/openshift-metal3/dev-scripts/pull/623

https://github.com/openshift-metal3/dev-scripts/issues/561 tracks the removal and the MCO PR is https://github.com/openshift/machine-config-operator/pull/795

Reply:
In dev scripts I suppose you could use an “oc adm release” command to inspect the release image and pull out the pullspec for an image in that release payload. That would just be a very short term hack though.